prometheus/exporter/intervalpush/exporter.go

```diff
@@ -16,6 +16,8 @@ const defaultInterval = 15 * time.Second
 type exporter struct {
 	t *time.Ticker
 	p *push.Pusher
+
+	done chan chan error
 }
 
 func NewExporter(url string, interval time.Duration) exp.Exporter {
```

Comment:
an interface is returned here  that does not include io.Closer

Reply:
As stated in the description, it requires to downcast to not change the interface.